wallet/src/main/scala/org/bitcoins/wallet/models/OutgoingTransactionTable.scala

```diff
@@ -0,0 +1,162 @@
+package org.bitcoins.wallet.models
+
+import org.bitcoins.core.crypto._
+import org.bitcoins.core.currency._
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.script.EmptyScriptSignature
+import org.bitcoins.core.protocol.transaction._
+import org.bitcoins.core.wallet.fee.{FeeUnit, SatoshisPerVirtualByte}
+import slick.jdbc.SQLiteProfile.api._
+import slick.lifted.{PrimaryKey, ProvenShape}
+
+/** Represents a relevant transaction for the wallet that we should be keeping track of */
+case class OutgoingTransactionDb(
+    txIdBE: DoubleSha256DigestBE,
+    transaction: Transaction,
+    unsignedTxIdBE: DoubleSha256DigestBE,
+    unsignedTx: Transaction,
+    numInputs: Int,
+    numOutputs: Int,
+    lockTime: UInt32,
+    txFee: CurrencyUnit,
+    expectedFeeRate: FeeUnit,
+    actualFeeRate: FeeUnit) {
+  require(unsignedTx.inputs.forall(_.scriptSignature == EmptyScriptSignature),
+          s"All ScriptSignatures must be empty, got $unsignedTx")
+
+  lazy val txId: DoubleSha256Digest = txIdBE.flip
+  lazy val unsignedTxId: DoubleSha256Digest = unsignedTxIdBE.flip
+}
+
+object OutgoingTransactionDb {
+
+  def fromTransaction(
+      tx: Transaction,
+      expectedFeeRate: FeeUnit,
+      inputAmount: CurrencyUnit): OutgoingTransactionDb = {
+    val feePaid = inputAmount - tx.outputs.map(_.value).sum
+    val feeRate = feePaid.satoshis.toLong / tx.vsize
+    val unsignedTx = tx match {
+      case btx: BaseTransaction =>
+        val unsignedInputs = btx.inputs.map(
+          input =>
+            TransactionInput(input.previousOutput,
+                             EmptyScriptSignature,
+                             input.sequence))
+        BaseTransaction(btx.version, unsignedInputs, btx.outputs, btx.lockTime)
+      case wtx: WitnessTransaction =>
+        val unsignedInputs = wtx.inputs.map(
+          input =>
+            TransactionInput(input.previousOutput,
+                             EmptyScriptSignature,
+                             input.sequence))
+        WitnessTransaction(wtx.version,
+                           unsignedInputs,
+                           wtx.outputs,
+                           wtx.lockTime,
+                           EmptyWitness.fromInputs(unsignedInputs))
+    }
+    OutgoingTransactionDb(tx.txIdBE,
+                          tx,
+                          unsignedTx.txIdBE,
+                          unsignedTx,
+                          tx.inputs.size,
+                          tx.outputs.size,
+                          tx.lockTime,
+                          feePaid,
+                          expectedFeeRate,
+                          SatoshisPerVirtualByte(feeRate))
+  }
+}
+
+class OutgoingTransactionTable(tag: Tag)
+    extends Table[OutgoingTransactionDb](tag, "wallet_outgoing_txs") {
+
```

Comment:
Shouldn't there be a foreign key to our utxo table with utxos we are using to fund these txs?

Reply:
To have an actual enforced foreign key, we would need a column for each out point. So we could do `outPoint0`, `outPoint1`, `outPoint2`, `outPoint3`, etc, but that is pretty ugly and we have to put a cap of inputs we can put in a tx